### ISSUE TITLE
[JENKINS-13869] [Unexpected termination of the channel] for Unix SSH slaves

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/remoting-gh-issues-poc-11-15#233]
<p>Once we upgraded to 1.464 jobs that run on Unix virtual machines (using SSH launch method) started to fail with the error below. The use case is the following:</p>

<p>We have a job that reverts the VM and triggers a parametrized build on install jobs passing the node label with NodeLabel parameter plugin. Nodes have In demand delay = 5 and Idle delay = 15. <br/>
Node os: Linux vmbam09 2.6.18-194.el5 #1 SMP Tue Mar 16 21:52:39 EDT 2010 x86_64 x86_64 x86_64 GNU/Linux</p>

<p>Console Output for the failing jobs:<br/>
Started by upstream project "eda-vm-revert" build number 4800<br/>
Building remotely on vmbam09.eur.ad.sag in workspace /home/vmtest/jenkins/workspace/eda-install<br/>
FATAL: hudson.remoting.RequestAbortedException: java.io.IOException: Unexpected termination of the channel<br/>
hudson.remoting.RequestAbortedException: hudson.remoting.RequestAbortedException: java.io.IOException: Unexpected termination of the channel<br/>
	at hudson.remoting.Request.call(Request.java:149)<br/>
	at hudson.remoting.Channel.call(Channel.java:646)<br/>
	at hudson.FilePath.act(FilePath.java:828)<br/>
	at hudson.FilePath.act(FilePath.java:821)<br/>
	at hudson.FilePath.mkdirs(FilePath.java:887)<br/>
	at hudson.model.AbstractProject.checkout(AbstractProject.java:1216)<br/>
	at hudson.model.AbstractBuild$AbstractRunner.checkout(AbstractBuild.java:586)<br/>
	at hudson.model.AbstractBuild$AbstractRunner.run(AbstractBuild.java:475)<br/>
	at hudson.model.Run.run(Run.java:1434)<br/>
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:46)<br/>
	at hudson.model.ResourceController.execute(ResourceController.java:88)<br/>
	at hudson.model.Executor.run(Executor.java:239)<br/>
Caused by: hudson.remoting.RequestAbortedException: java.io.IOException: Unexpected termination of the channel<br/>
	at hudson.remoting.Request.abort(Request.java:273)<br/>
	at hudson.remoting.Channel.terminate(Channel.java:702)<br/>
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:69)<br/>
Caused by: java.io.IOException: Unexpected termination of the channel<br/>
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:50)<br/>
Caused by: java.io.EOFException<br/>
	at java.io.ObjectInputStream$BlockDataInputStream.peekByte(ObjectInputStream.java:2553)<br/>
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1296)<br/>
	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:350)<br/>
	at hudson.remoting.Command.readFrom(Command.java:90)<br/>
	at hudson.remoting.ClassicCommandTransport.read(ClassicCommandTransport.java:59)<br/>
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:48)</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vassilena">vassilena</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-13869" target="_blank">[Unexpected termination of the channel] for Unix SSH slaves</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kohsuke.png" title="kohsuke's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kohsuke">kohsuke</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Critical
<li><b>resolution</b>: Cannot Reproduce
<li><b>resolved</b>: 2014-10-06T21:46:20+00:00
<li><b>votes</b>: 15
<li><b>watchers</b>: 16
<li><b>imported</b>: 2025-11-15
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Once we upgraded to 1.464 jobs that run on Unix virtual machines (using SSH launch method) started to fail with the error below. The use case is the following:</p>

<p>We have a job that reverts the VM and triggers a parametrized build on install jobs passing the node label with NodeLabel parameter plugin. Nodes have In demand delay = 5 and Idle delay = 15. 
Node os: Linux vmbam09 2.6.18-194.el5 #1 SMP Tue Mar 16 21:52:39 EDT 2010 x86_64 x86_64 x86_64 GNU/Linux</p>

<p>Console Output for the failing jobs:
Started by upstream project "eda-vm-revert" build number 4800
Building remotely on vmbam09.eur.ad.sag in workspace /home/vmtest/jenkins/workspace/eda-install
FATAL: hudson.remoting.RequestAbortedException: java.io.IOException: Unexpected termination of the channel
hudson.remoting.RequestAbortedException: hudson.remoting.RequestAbortedException: java.io.IOException: Unexpected termination of the channel
	at hudson.remoting.Request.call(Request.java:149)
	at hudson.remoting.Channel.call(Channel.java:646)
	at hudson.FilePath.act(FilePath.java:828)
	at hudson.FilePath.act(FilePath.java:821)
	at hudson.FilePath.mkdirs(FilePath.java:887)
	at hudson.model.AbstractProject.checkout(AbstractProject.java:1216)
	at hudson.model.AbstractBuild$AbstractRunner.checkout(AbstractBuild.java:586)
	at hudson.model.AbstractBuild$AbstractRunner.run(AbstractBuild.java:475)
	at hudson.model.Run.run(Run.java:1434)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:46)
	at hudson.model.ResourceController.execute(ResourceController.java:88)
	at hudson.model.Executor.run(Executor.java:239)
Caused by: hudson.remoting.RequestAbortedException: java.io.IOException: Unexpected termination of the channel
	at hudson.remoting.Request.abort(Request.java:273)
	at hudson.remoting.Channel.terminate(Channel.java:702)
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:69)
Caused by: java.io.IOException: Unexpected termination of the channel
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:50)
Caused by: java.io.EOFException
	at java.io.ObjectInputStream$BlockDataInputStream.peekByte(ObjectInputStream.java:2553)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1296)
	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:350)
	at hudson.remoting.Command.readFrom(Command.java:90)
	at hudson.remoting.ClassicCommandTransport.read(ClassicCommandTransport.java:59)
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:48)</p></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>Jenkins Master version: 1.464 hosted on Linux bambpmsauto 2.6.34.7-0.4-desktop #1 SMP PREEMPT 2010-10-07 19:07:51 +0200 x86_64 x86_64 x86_64 GNU/Linux</code></li></ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>gruello's <a href="https://issues.jenkins.io/browse/JENKINS-13869?focusedId=165200&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-165200">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Hi, do we foreseen when the ticket will be closed?</p>
</pre>
</details>

</details>
<p>Hi, do we foreseen when the ticket will be closed?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>hiteswar_kumar's <a href="https://issues.jenkins.io/browse/JENKINS-13869?focusedId=181067&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-181067">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>this issue still at ssh-slave .22 and Jnekins core lts 1.480.3
please share comment to resolve this issue.</p>
</pre>
</details>

</details>
<p>this issue still at ssh-slave .22 and Jnekins core lts 1.480.3<br/>
please share comment to resolve this issue.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ssbarnea.png" title="ssbarnea's avatar" /> ssbarnea's <a href="https://issues.jenkins.io/browse/JENKINS-13869?focusedId=181357&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-181357">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>I am not sure if it is the same problem but I get:</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">Jul 10, 2013 5:45:33 PM hudson.remoting.SynchronousCommandTransport$ReaderThread run
SEVERE: I/O error in channel miko
java.io.IOException: Unexpected termination of the channel
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:50)
Caused by: java.io.EOFException
	at java.io.ObjectInputStream$BlockDataInputStream.peekByte(ObjectInputStream.java:2596)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1316)
	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:370)
	at hudson.remoting.Command.readFrom(Command.java:92)
	at hudson.remoting.ClassicCommandTransport.read(ClassicCommandTransport.java:72)
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:48)

Jul 10, 2013 5:45:33 PM hudson.remoting.SynchronousCommandTransport$ReaderThread run
</pre>
</div></div>

<p>Currently all slaves are down due to this.</p>
</pre>
</details>

</details>
<p>I am not sure if it is the same problem but I get:</p>

<pre>
Jul 10, 2013 5:45:33 PM hudson.remoting.SynchronousCommandTransport$ReaderThread run
SEVERE: I/O error in channel miko
java.io.IOException: Unexpected termination of the channel
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:50)
Caused by: java.io.EOFException
	at java.io.ObjectInputStream$BlockDataInputStream.peekByte(ObjectInputStream.java:2596)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1316)
	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:370)
	at hudson.remoting.Command.readFrom(Command.java:92)
	at hudson.remoting.ClassicCommandTransport.read(ClassicCommandTransport.java:72)
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:48)

Jul 10, 2013 5:45:33 PM hudson.remoting.SynchronousCommandTransport$ReaderThread run
</pre>

<p>Currently all slaves are down due to this.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ssbarnea.png" title="ssbarnea's avatar" /> ssbarnea's <a href="https://issues.jenkins.io/browse/JENKINS-13869?focusedId=181360&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-181360">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Initially I had the impression that the problem was the SSH keys but here is the proof that SSH works and that jenkins server is lcosing the connection 5 seconds lates. Nothing more logged on the server side.</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">Jul 10 18:30:07 sorins sshd[4067]: reverse mapping checking getaddrinfo <span class="code-keyword">for</span> dote-02.80.10.in-addr.arpa [10.80.225.152] failed - POSSIBLE BREAK-IN ATTEMPT!
Jul 10 18:30:08 sorins sshd[4067]: Accepted publickey <span class="code-keyword">for</span> xenbuild from 10.80.225.152 port 40826 ssh2
Jul 10 18:30:08 sorins sshd[4067]: pam_unix(sshd:session): session opened <span class="code-keyword">for</span> user xenbuild by (uid=0)
Jul 10 18:30:08 sorins sshd[4214]: subsystem request <span class="code-keyword">for</span> sftp by user xenbuild
Jul 10 18:30:13 sorins sshd[4214]: Received disconnect from 10.80.225.152: 11: Closed due to user request.
Jul 10 18:30:13 sorins sshd[4067]: pam_unix(sshd:session): session closed <span class="code-keyword">for</span> user xenbuild
</pre>
</div></div>
</pre>
</details>

</details>
<p>Initially I had the impression that the problem was the SSH keys but here is the proof that SSH works and that jenkins server is lcosing the connection 5 seconds lates. Nothing more logged on the server side.</p>

<pre>
Jul 10 18:30:07 sorins sshd[4067]: reverse mapping checking getaddrinfo <span class="code-keyword">for</span> dote-02.80.10.in-addr.arpa [10.80.225.152] failed - POSSIBLE BREAK-IN ATTEMPT!
Jul 10 18:30:08 sorins sshd[4067]: Accepted publickey <span class="code-keyword">for</span> xenbuild from 10.80.225.152 port 40826 ssh2
Jul 10 18:30:08 sorins sshd[4067]: pam_unix(sshd:session): session opened <span class="code-keyword">for</span> user xenbuild by (uid=0)
Jul 10 18:30:08 sorins sshd[4214]: subsystem request <span class="code-keyword">for</span> sftp by user xenbuild
Jul 10 18:30:13 sorins sshd[4214]: Received disconnect from 10.80.225.152: 11: Closed due to user request.
Jul 10 18:30:13 sorins sshd[4067]: pam_unix(sshd:session): session closed <span class="code-keyword">for</span> user xenbuild
</pre>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>tlallart's <a href="https://issues.jenkins.io/browse/JENKINS-13869?focusedId=181442&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-181442">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Hi</p>

<p>I have 3 jenkins Slaves connecting with SSHSlaveAgent and private key : 2 RedHat 6 Linux nodes and 1 Solaris X node.
Updating Jenkins from 1.514 to 1.522, I got now on my Solaris node :</p>

<p>java.io.IOException: Unexpected termination of the channel
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:50)
Caused by: java.io.EOFException
	at java.io.ObjectInputStream$BlockDataInputStream.peekByte(ObjectInputStream.java:2571)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1315)
	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:369)
	at hudson.remoting.Command.readFrom(Command.java:92)
	at hudson.remoting.ClassicCommandTransport.read(ClassicCommandTransport.java:72)
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:48)</p>

<p>Connecting directly from command line with ssh user@host works well.</p>

<p>My Linux nodes are connected by Jenkins say that clock are not synchronized, 1 day and 1 hour late. However, I'm pretty sure that my master and this nodes have the same time synchronized by a ntp server.</p>
</pre>
</details>

</details>
<p>Hi</p>

<p>I have 3 jenkins Slaves connecting with SSHSlaveAgent and private key : 2 RedHat 6 Linux nodes and 1 Solaris X node.<br/>
Updating Jenkins from 1.514 to 1.522, I got now on my Solaris node :</p>

<p>java.io.IOException: Unexpected termination of the channel<br/>
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:50)<br/>
Caused by: java.io.EOFException<br/>
	at java.io.ObjectInputStream$BlockDataInputStream.peekByte(ObjectInputStream.java:2571)<br/>
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1315)<br/>
	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:369)<br/>
	at hudson.remoting.Command.readFrom(Command.java:92)<br/>
	at hudson.remoting.ClassicCommandTransport.read(ClassicCommandTransport.java:72)<br/>
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:48)</p>

<p>Connecting directly from command line with ssh user@host works well.</p>

<p>My Linux nodes are connected by Jenkins say that clock are not synchronized, 1 day and 1 hour late. However, I'm pretty sure that my master and this nodes have the same time synchronized by a ntp server.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ssbarnea.png" title="ssbarnea's avatar" /> ssbarnea's <a href="https://issues.jenkins.io/browse/JENKINS-13869?focusedId=181462&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-181462">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>I was able to fix my problem, after spending many hours looking for a way to increase the log level of jenkins. In my case the cause was that the slave setup script was failing, that's added by an optional which runs some commands before starting the slave.jar Still, it seems that jenkins fails ugly when this happens.</p>

<p>Another cause was that the slave was not able to run because the build directory was not writable (after swiching from using root to using a different account, obviously the existing files were owned by root). Still, same ugly behaviour and nothing useful in the console.</p>

<p>Also, I still have the same problem regarding the clock, which is in sync but jenkins is reporting even 10 days behind on one of the slaves. </p>
</pre>
</details>

</details>
<p>I was able to fix my problem, after spending many hours looking for a way to increase the log level of jenkins. In my case the cause was that the slave setup script was failing, that's added by an optional which runs some commands before starting the slave.jar Still, it seems that jenkins fails ugly when this happens.</p>

<p>Another cause was that the slave was not able to run because the build directory was not writable (after swiching from using root to using a different account, obviously the existing files were owned by root). Still, same ugly behaviour and nothing useful in the console.</p>

<p>Also, I still have the same problem regarding the clock, which is in sync but jenkins is reporting even 10 days behind on one of the slaves. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/hgomez.png" title="hgomez's avatar" /> hgomez's <a href="https://issues.jenkins.io/browse/JENKINS-13869?focusedId=185655&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-185655">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Same problem here.</p>

<p>I notice it happen more and more frequently since I upgraded Jenkins to 1.529.
All plugins up to date.</p>

<p>ie :</p>


<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">FATAL: Unable to delete script file /tmp/hudson6897412732837401550.sh
hudson.util.IOException2: remote file operation failed: /tmp/hudson6897412732837401550.sh at hudson.remoting.Channel@3670a980:devops-builder-c632
	at hudson.FilePath.act(FilePath.java:905)
	at hudson.FilePath.act(FilePath.java:882)
	at hudson.FilePath.delete(FilePath.java:1291)
	at hudson.tasks.CommandInterpreter.perform(CommandInterpreter.java:101)
	at hudson.tasks.CommandInterpreter.perform(CommandInterpreter.java:60)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:782)
	at hudson.model.Build$BuildExecution.build(Build.java:199)
	at hudson.model.Build$BuildExecution.doRun(Build.java:160)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:567)
	at hudson.model.Run.execute(Run.java:1603)
	at hudson.matrix.MatrixRun.run(MatrixRun.java:146)
	at hudson.model.ResourceController.execute(ResourceController.java:88)
	at hudson.model.Executor.run(Executor.java:246)
Caused by: hudson.remoting.ChannelClosedException: channel is already closed
	at hudson.remoting.Channel.send(Channel.java:516)
	at hudson.remoting.Request.call(Request.java:129)
	at hudson.remoting.Channel.call(Channel.java:714)
	at hudson.FilePath.act(FilePath.java:898)
	... 13 more
Caused by: java.io.IOException: Unexpected termination of the channel
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:50)
Caused by: java.io.EOFException
	at java.io.ObjectInputStream$BlockDataInputStream.peekByte(ObjectInputStream.java:2576)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1295)
	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:349)
	at hudson.remoting.Command.readFrom(Command.java:92)
	at hudson.remoting.ClassicCommandTransport.read(ClassicCommandTransport.java:71)
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:48)
FATAL: hudson.remoting.RequestAbortedException: java.io.IOException: Unexpected termination of the channel
hudson.remoting.RequestAbortedException: hudson.remoting.RequestAbortedException: java.io.IOException: Unexpected termination of the channel
	at hudson.remoting.RequestAbortedException.wrapForRethrow(RequestAbortedException.java:41)
	at hudson.remoting.RequestAbortedException.wrapForRethrow(RequestAbortedException.java:34)
	at hudson.remoting.Request.call(Request.java:174)
	at hudson.remoting.Channel.call(Channel.java:714)
	at hudson.remoting.RemoteInvocationHandler.invoke(RemoteInvocationHandler.java:167)
	at com.sun.proxy.$Proxy74.join(Unknown Source)
	at hudson.Launcher$RemoteLauncher$ProcImpl.join(Launcher.java:925)
	at hudson.Launcher$ProcStarter.join(Launcher.java:360)
	at hudson.tasks.CommandInterpreter.perform(CommandInterpreter.java:91)
	at hudson.tasks.CommandInterpreter.perform(CommandInterpreter.java:60)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:782)
	at hudson.model.Build$BuildExecution.build(Build.java:199)
	at hudson.model.Build$BuildExecution.doRun(Build.java:160)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:567)
	at hudson.model.Run.execute(Run.java:1603)
	at hudson.matrix.MatrixRun.run(MatrixRun.java:146)
	at hudson.model.ResourceController.execute(ResourceController.java:88)
	at hudson.model.Executor.run(Executor.java:246)
Caused by: hudson.remoting.RequestAbortedException: java.io.IOException: Unexpected termination of the channel
	at hudson.remoting.Request.abort(Request.java:299)
	at hudson.remoting.Channel.terminate(Channel.java:774)
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:69)
Caused by: java.io.IOException: Unexpected termination of the channel
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:50)
Caused by: java.io.EOFException
	at java.io.ObjectInputStream$BlockDataInputStream.peekByte(ObjectInputStream.java:2576)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1295)
	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:349)
	at hudson.remoting.Command.readFrom(Command.java:92)
	at hudson.remoting.ClassicCommandTransport.read(ClassicCommandTransport.java:71)
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:48)
</pre>
</div></div>

<p>In this sample, slaves are VBox VMs, hosted on same machine, and provisioned by Jenkins VirtualBox Plugin.
Weird</p>

</pre>
</details>

</details>
<p>Same problem here.</p>

<p>I notice it happen more and more frequently since I upgraded Jenkins to 1.529.<br/>
All plugins up to date.</p>

<p>ie :</p>



<pre>
FATAL: Unable to delete script file /tmp/hudson6897412732837401550.sh
hudson.util.IOException2: remote file operation failed: /tmp/hudson6897412732837401550.sh at hudson.remoting.Channel@3670a980:devops-builder-c632
	at hudson.FilePath.act(FilePath.java:905)
	at hudson.FilePath.act(FilePath.java:882)
	at hudson.FilePath.delete(FilePath.java:1291)
	at hudson.tasks.CommandInterpreter.perform(CommandInterpreter.java:101)
	at hudson.tasks.CommandInterpreter.perform(CommandInterpreter.java:60)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:782)
	at hudson.model.Build$BuildExecution.build(Build.java:199)
	at hudson.model.Build$BuildExecution.doRun(Build.java:160)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:567)
	at hudson.model.Run.execute(Run.java:1603)
	at hudson.matrix.MatrixRun.run(MatrixRun.java:146)
	at hudson.model.ResourceController.execute(ResourceController.java:88)
	at hudson.model.Executor.run(Executor.java:246)
Caused by: hudson.remoting.ChannelClosedException: channel is already closed
	at hudson.remoting.Channel.send(Channel.java:516)
	at hudson.remoting.Request.call(Request.java:129)
	at hudson.remoting.Channel.call(Channel.java:714)
	at hudson.FilePath.act(FilePath.java:898)
	... 13 more
Caused by: java.io.IOException: Unexpected termination of the channel
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:50)
Caused by: java.io.EOFException
	at java.io.ObjectInputStream$BlockDataInputStream.peekByte(ObjectInputStream.java:2576)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1295)
	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:349)
	at hudson.remoting.Command.readFrom(Command.java:92)
	at hudson.remoting.ClassicCommandTransport.read(ClassicCommandTransport.java:71)
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:48)
FATAL: hudson.remoting.RequestAbortedException: java.io.IOException: Unexpected termination of the channel
hudson.remoting.RequestAbortedException: hudson.remoting.RequestAbortedException: java.io.IOException: Unexpected termination of the channel
	at hudson.remoting.RequestAbortedException.wrapForRethrow(RequestAbortedException.java:41)
	at hudson.remoting.RequestAbortedException.wrapForRethrow(RequestAbortedException.java:34)
	at hudson.remoting.Request.call(Request.java:174)
	at hudson.remoting.Channel.call(Channel.java:714)
	at hudson.remoting.RemoteInvocationHandler.invoke(RemoteInvocationHandler.java:167)
	at com.sun.proxy.$Proxy74.join(Unknown Source)
	at hudson.Launcher$RemoteLauncher$ProcImpl.join(Launcher.java:925)
	at hudson.Launcher$ProcStarter.join(Launcher.java:360)
	at hudson.tasks.CommandInterpreter.perform(CommandInterpreter.java:91)
	at hudson.tasks.CommandInterpreter.perform(CommandInterpreter.java:60)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:782)
	at hudson.model.Build$BuildExecution.build(Build.java:199)
	at hudson.model.Build$BuildExecution.doRun(Build.java:160)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:567)
	at hudson.model.Run.execute(Run.java:1603)
	at hudson.matrix.MatrixRun.run(MatrixRun.java:146)
	at hudson.model.ResourceController.execute(ResourceController.java:88)
	at hudson.model.Executor.run(Executor.java:246)
Caused by: hudson.remoting.RequestAbortedException: java.io.IOException: Unexpected termination of the channel
	at hudson.remoting.Request.abort(Request.java:299)
	at hudson.remoting.Channel.terminate(Channel.java:774)
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:69)
Caused by: java.io.IOException: Unexpected termination of the channel
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:50)
Caused by: java.io.EOFException
	at java.io.ObjectInputStream$BlockDataInputStream.peekByte(ObjectInputStream.java:2576)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1295)
	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:349)
	at hudson.remoting.Command.readFrom(Command.java:92)
	at hudson.remoting.ClassicCommandTransport.read(ClassicCommandTransport.java:71)
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:48)
</pre>

<p>In this sample, slaves are VBox VMs, hosted on same machine, and provisioned by Jenkins VirtualBox Plugin.<br/>
Weird</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/nrc.png" title="nrc's avatar" /> nrc's <a href="https://issues.jenkins.io/browse/JENKINS-13869?focusedId=190472&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-190472">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Hi everyone. 
Is any development continued on this issue? We are getting these quite often and JVM tuning doesn't help <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/sad.png" height="16" width="16" align="absmiddle" alt="" border="0"/> any plans when it could be fixed?</p>
</pre>
</details>

</details>
<p>Hi everyone. <br/>
Is any development continued on this issue? We are getting these quite often and JVM tuning doesn't help <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/sad.png" height="16" width="16" align="absmiddle" alt="" border="0"/> any plans when it could be fixed?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/hgomez.png" title="hgomez's avatar" /> hgomez's <a href="https://issues.jenkins.io/browse/JENKINS-13869?focusedId=190473&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-190473">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>It seems this issue has been fixed in Jenkins 1.536 :</p>

<p>See <a href="https://issues.jenkins.io/browse/JENKINS-18836" title="java.io.IOException: Unexpected termination of the channel" class="issue-link" data-issue-key="JENKINS-18836"><del>JENKINS-18836</del></a>, <a href="https://issues.jenkins.io/browse/JENKINS-18879" title="Collecting finbugs analysis results randomly fails with exception" class="issue-link" data-issue-key="JENKINS-18879"><del>JENKINS-18879</del></a> and <a href="https://issues.jenkins.io/browse/JENKINS-19619" title="Collecting findbugs analysis results occasionally causes ssh slave to go offline causing job to abort" class="issue-link" data-issue-key="JENKINS-19619"><del>JENKINS-19619</del></a></p>
</pre>
</details>

</details>
<p>It seems this issue has been fixed in Jenkins 1.536 :</p>

<p>See <a href="https://issues.jenkins.io/browse/JENKINS-18836" title="java.io.IOException: Unexpected termination of the channel" class="issue-link" data-issue-key="JENKINS-18836"><del>JENKINS-18836</del></a>, <a href="https://issues.jenkins.io/browse/JENKINS-18879" title="Collecting finbugs analysis results randomly fails with exception" class="issue-link" data-issue-key="JENKINS-18879"><del>JENKINS-18879</del></a> and <a href="https://issues.jenkins.io/browse/JENKINS-19619" title="Collecting findbugs analysis results occasionally causes ssh slave to go offline causing job to abort" class="issue-link" data-issue-key="JENKINS-19619"><del>JENKINS-19619</del></a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/nrc.png" title="nrc's avatar" /> nrc's <a href="https://issues.jenkins.io/browse/JENKINS-13869?focusedId=190481&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-190481">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Hi, last Friday we upgraded to 1.540. Indeed, situation seems better but still we got one build failed with following error:</p>

<p>10:49:07 FATAL: hudson.remoting.RequestAbortedException: hudson.remoting.Channel$OrderlyShutdown
10:49:07 hudson.remoting.RequestAbortedException: hudson.remoting.RequestAbortedException: hudson.remoting.Channel$OrderlyShutdown
10:49:07 	at hudson.remoting.RequestAbortedException.wrapForRethrow(RequestAbortedException.java:41)
10:49:07 	at hudson.remoting.RequestAbortedException.wrapForRethrow(RequestAbortedException.java:34)
10:49:07 	at hudson.remoting.Request.call(Request.java:174)
10:49:07 	at hudson.remoting.Channel.call(Channel.java:722)
10:49:07 	at hudson.remoting.RemoteInvocationHandler.invoke(RemoteInvocationHandler.java:167)
10:49:07 	at $Proxy55.join(Unknown Source)
10:49:07 	at hudson.Launcher$RemoteLauncher$ProcImpl.join(Launcher.java:925)
10:49:07 	at hudson.Launcher$ProcStarter.join(Launcher.java:360)
10:49:07 	at hudson.tasks.CommandInterpreter.perform(CommandInterpreter.java:94)
10:49:07 	at hudson.tasks.CommandInterpreter.perform(CommandInterpreter.java:63)
10:49:07 	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
10:49:07 	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:785)
10:49:07 	at hudson.model.Build$BuildExecution.build(Build.java:199)
10:49:07 	at hudson.model.Build$BuildExecution.doRun(Build.java:160)
10:49:07 	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:566)
10:49:07 	at hudson.model.Run.execute(Run.java:1679)
10:49:07 	at hudson.matrix.MatrixRun.run(MatrixRun.java:146)
10:49:07 	at hudson.model.ResourceController.execute(ResourceController.java:88)
10:49:07 	at hudson.model.Executor.run(Executor.java:230)
10:49:07 Caused by: hudson.remoting.RequestAbortedException: hudson.remoting.Channel$OrderlyShutdown
10:49:07 	at hudson.remoting.Request.abort(Request.java:299)
10:49:07 	at hudson.remoting.Channel.terminate(Channel.java:782)
10:49:07 	at hudson.remoting.Channel$CloseCommand.execute(Channel.java:927)
10:49:07 	at hudson.remoting.Channel$2.handle(Channel.java:461)
10:49:07 	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:60)
10:49:07 Caused by: hudson.remoting.Channel$OrderlyShutdown
10:49:07 	... 3 more
10:49:07 Caused by: Command close created at
10:49:07 	at hudson.remoting.Command.&lt;init&gt;(Command.java:56)
10:49:07 	at hudson.remoting.Channel$CloseCommand.&lt;init&gt;(Channel.java:921)
10:49:07 	at hudson.remoting.Channel$CloseCommand.&lt;init&gt;(Channel.java:919)
10:49:07 	at hudson.remoting.Channel.close(Channel.java:1002)
10:49:07 	at hudson.remoting.Channel.close(Channel.java:985)
10:49:07 	at hudson.remoting.Channel$CloseCommand.execute(Channel.java:926)
10:49:07 	... 2 more</p>
</pre>
</details>

</details>
<p>Hi, last Friday we upgraded to 1.540. Indeed, situation seems better but still we got one build failed with following error:</p>

<p>10:49:07 FATAL: hudson.remoting.RequestAbortedException: hudson.remoting.Channel$OrderlyShutdown<br/>
10:49:07 hudson.remoting.RequestAbortedException: hudson.remoting.RequestAbortedException: hudson.remoting.Channel$OrderlyShutdown<br/>
10:49:07 	at hudson.remoting.RequestAbortedException.wrapForRethrow(RequestAbortedException.java:41)<br/>
10:49:07 	at hudson.remoting.RequestAbortedException.wrapForRethrow(RequestAbortedException.java:34)<br/>
10:49:07 	at hudson.remoting.Request.call(Request.java:174)<br/>
10:49:07 	at hudson.remoting.Channel.call(Channel.java:722)<br/>
10:49:07 	at hudson.remoting.RemoteInvocationHandler.invoke(RemoteInvocationHandler.java:167)<br/>
10:49:07 	at $Proxy55.join(Unknown Source)<br/>
10:49:07 	at hudson.Launcher$RemoteLauncher$ProcImpl.join(Launcher.java:925)<br/>
10:49:07 	at hudson.Launcher$ProcStarter.join(Launcher.java:360)<br/>
10:49:07 	at hudson.tasks.CommandInterpreter.perform(CommandInterpreter.java:94)<br/>
10:49:07 	at hudson.tasks.CommandInterpreter.perform(CommandInterpreter.java:63)<br/>
10:49:07 	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)<br/>
10:49:07 	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:785)<br/>
10:49:07 	at hudson.model.Build$BuildExecution.build(Build.java:199)<br/>
10:49:07 	at hudson.model.Build$BuildExecution.doRun(Build.java:160)<br/>
10:49:07 	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:566)<br/>
10:49:07 	at hudson.model.Run.execute(Run.java:1679)<br/>
10:49:07 	at hudson.matrix.MatrixRun.run(MatrixRun.java:146)<br/>
10:49:07 	at hudson.model.ResourceController.execute(ResourceController.java:88)<br/>
10:49:07 	at hudson.model.Executor.run(Executor.java:230)<br/>
10:49:07 Caused by: hudson.remoting.RequestAbortedException: hudson.remoting.Channel$OrderlyShutdown<br/>
10:49:07 	at hudson.remoting.Request.abort(Request.java:299)<br/>
10:49:07 	at hudson.remoting.Channel.terminate(Channel.java:782)<br/>
10:49:07 	at hudson.remoting.Channel$CloseCommand.execute(Channel.java:927)<br/>
10:49:07 	at hudson.remoting.Channel$2.handle(Channel.java:461)<br/>
10:49:07 	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:60)<br/>
10:49:07 Caused by: hudson.remoting.Channel$OrderlyShutdown<br/>
10:49:07 	... 3 more<br/>
10:49:07 Caused by: Command close created at<br/>
10:49:07 	at hudson.remoting.Command.<init>(Command.java:56)<br/>
10:49:07 	at hudson.remoting.Channel$CloseCommand.<init>(Channel.java:921)<br/>
10:49:07 	at hudson.remoting.Channel$CloseCommand.<init>(Channel.java:919)<br/>
10:49:07 	at hudson.remoting.Channel.close(Channel.java:1002)<br/>
10:49:07 	at hudson.remoting.Channel.close(Channel.java:985)<br/>
10:49:07 	at hudson.remoting.Channel$CloseCommand.execute(Channel.java:926)<br/>
10:49:07 	... 2 more</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-13869?focusedId=213014&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-213014">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Given the age and nature of this issue and the comments indicating this is partly fixed or actually several issues, I'm resolving this as Cannot Reproduce.</p>

<p>This doesn't mean the issue reported here isn't valid or anything like that. But Jenkins has been improved a lot in the mean time, so the value of this issue report is questionable by now (is it still relevant? Who knows?). It seems this slipped through the cracks as it was filed against a completely unrelated component (the Slave Setup Plugin).</p>

<p>If you encounter similar issues in then-recent Jenkins versions (say, no older than eight weeks), <b>please file a new issue</b> against the <b>remoting</b> component following the instructions here: <a href="https://wiki.jenkins-ci.org/display/JENKINS/How+to+report+an+issue" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/How+to+report+an+issue</a></p>
</pre>
</details>

</details>
<p>Given the age and nature of this issue and the comments indicating this is partly fixed or actually several issues, I'm resolving this as Cannot Reproduce.</p>

<p>This doesn't mean the issue reported here isn't valid or anything like that. But Jenkins has been improved a lot in the mean time, so the value of this issue report is questionable by now (is it still relevant? Who knows?). It seems this slipped through the cracks as it was filed against a completely unrelated component (the Slave Setup Plugin).</p>

<p>If you encounter similar issues in then-recent Jenkins versions (say, no older than eight weeks), <b>please file a new issue</b> against the <b>remoting</b> component following the instructions here: <a href="https://wiki.jenkins-ci.org/display/JENKINS/How+to+report+an+issue" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/How+to+report+an+issue</a></p>
